### PR TITLE
[gcp-deployer] add service account key handler and a check health handler

### PR DESCRIPTION
**Pull request by kunmingg**
Add 2 handlers to web app go backend:

service account key handler: 
given service account and tmp token, create service account key and insert into GKE cluster.

check health handler
check if backend respond to post request.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/1329)

### CODE REVIEW

File: bootstrap/cmd/bootstrap/app/ksServer.go [hunk 1]
```diff
@@ -534,6 +566,7 @@ func decodeCreateAppRequest(_ context.Context, r *http.Request) (interface{}, er
 
 // The same encoder can be used for all RPC responses.
 func encodeResponse(_ context.Context, w http.ResponseWriter, response interface{}) error {
+	log.Info("encode response: ")
```
Comment (yebrahim) [inline bootstrap/cmd/bootstrap/app/ksServer.go]: Did you meant to add this? It'll be very noisy.

File: bootstrap/cmd/bootstrap/app/ksServer.go [hunk 2]
```diff
@@ -550,6 +583,35 @@ func (s *ksServer) StartHttp(port int) {
...
+			if err := json.NewDecoder(r.Body).Decode(&request); err != nil {
+				return nil, err
+			}
+			return request, nil
+		},
+		encodeResponse,
+	)
+
 	http.Handle("/apps/create", createAppHandler)
+	http.Handle("/liveness", livenessHandler)
```
Comment (yebrahim) [inline bootstrap/cmd/bootstrap/app/ksServer.go]: Use `/healthz`?

Comment (yebrahim) [reply bootstrap/cmd/bootstrap/app/ksServer.go]: Similarly, can `makeLivenessEndpoint` be `makeHealthzEndpoint`.. etc?

File: bootstrap/cmd/bootstrap/app/ksServer.go [hunk 3]
```diff
@@ -515,8 +525,30 @@ func makeCreateAppEndpoint(svc KsService) endpoint.Endpoint {
...
+		}
+		return r, nil
+	}
+}
+
+func makeLivenessEndpoint(svc KsService) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(LiveRequest)
+		r := &LiveResponse{}
+		r.Reply = req.Msg + "accepted! Sill alive!"
```
Comment (yebrahim) [inline bootstrap/cmd/bootstrap/app/ksServer.go]: Are there no other checks we want to perform here to make sure the entire system is up and running? If the endpoint returns a success code, can the caller then assume that all other services are up.. etc?

File: bootstrap/cmd/bootstrap/app/SaKey.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+	})
+	k8sConfig, err := buildClusterConfig(ctx, ts, request)
+	if err != nil {
+		log.Errorf("Failed getting GKE cluster info: %v", err)
+		return err
+	}
+
+	c, err := iamadmin.NewIamClient(ctx, option.WithTokenSource(ts))
+	if err != nil {
+		log.Errorf("Can now create iam admin client: %v", err)
```
Comment (yebrahim) [inline bootstrap/cmd/bootstrap/app/SaKey.go]: "Cannot"*

File: bootstrap/cmd/bootstrap/app/SaKey.go [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+	secretData := make(map[string][]byte)
+	secretData[request.SecretKey] = createdKey.PrivateKeyData
+	_, err = k8sClientset.CoreV1().Secrets(request.Namespace).Create(
+		&v1.Secret{
+			ObjectMeta: meta_v1.ObjectMeta{
+				Namespace: request.Namespace,
+				Name:      request.SecretName,
+			},
+			Data: secretData,
+		})
```
Comment (yebrahim) [inline bootstrap/cmd/bootstrap/app/SaKey.go]: Is this call synchronous? What does the caller of the SA key endpoint expect to happen after making the API call? Do they then have to wait or poll somehow if they want to block on this operation?

**Comment by jlewi** [comment]
I took a look; I don't have anything to add beyond @yebrahim's comments.

File: bootstrap/cmd/bootstrap/app/ksServer.go (hunk 3 above)
Comment (kunmingg) [reply bootstrap/cmd/bootstrap/app/ksServer.go]: Currently this one only indicate backend is reachable and have cycles to respond to requests.

We can keep a server status variable to indicate backend status, which should kill the pod if needed so we have auto-start.
Let's create issue for it.

File: bootstrap/cmd/bootstrap/app/SaKey.go (hunk 5 above)
Comment (kunmingg) [reply bootstrap/cmd/bootstrap/app/SaKey.go]: It's synchronous. When handler returns, either the sa key already inserted into GKE or an error happened and included in response.

**Comment by yebrahim** [comment]
Awesome, thanks!
/lgtm

**Pull request merged**